src/libos/src/net/socket/socket_address.rs

```diff
@@ -0,0 +1,121 @@
+use super::*;
+use std::*;
+
+#[derive(Copy, Clone)]
+pub struct SockAddr {
+    storage: libc::sockaddr_storage,
+    len: usize,
+}
+
+// TODO: add more fields
+impl fmt::Debug for SockAddr {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SockAddr")
+            .field(
+                "family",
+                &AddressFamily::try_from(self.storage.ss_family).unwrap(),
+            )
+            .field("len", &self.len)
+            .finish()
+    }
+}
+
+impl SockAddr {
+    // Caller should guarentee the sockaddr and addr_len are valid
+    pub unsafe fn try_from_raw(
+        sockaddr: *const libc::sockaddr,
+        addr_len: libc::socklen_t,
+    ) -> Result<Self> {
```

Comment:
I cannot find any code that uses this method. Could you explain?

Reply:
It is used to transform `libc::sockaddr` to Rust `SockAddr`. You can find the usage in `src/libos/src/net/syscalls.rs`.

Reply:
Ok. I see. Thanks.